src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import {Location} from 'history';
+import styled from 'react-emotion';
+import {browserHistory} from 'react-router';
+
+import space from 'app/styles/space';
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import Button from 'app/components/button';
+import {Organization} from 'app/types';
+import {
+  deleteSavedQuery,
+  updateSavedQuery,
+} from 'app/actionCreators/discoverSavedQueries';
+import {SavedQuery} from 'app/stores/discoverSavedQueriesStore';
+import withApi from 'app/utils/withApi';
+import withDiscoverSavedQueries from 'app/utils/withDiscoverSavedQueries';
+import {addSuccessMessage} from 'app/actionCreators/indicator';
+
+import EventView from './eventView';
+import EventsSaveQueryButton from './saveQueryButton';
+
+type Props = {
+  api: Client;
+  eventView: EventView;
+  location: Location;
+  organization: Organization;
+  savedQueries: SavedQuery[];
+};
+
+class SavedQueryButtonGroup extends React.Component<Props> {
+  getExistingSavedQuery = (): EventView | undefined => {
```

Comment:
Is it better to use `null` or `undefined` in 'thing or nothing' style methods in typescript?

Reply:
Either `null` or `undefined` works fine. 

We've initially configured TypeScript to treat `null` and `undefined` as two different distinct values.

Reply:
Ok.

Reply:
I prefer `undefined` because `typeof null == "object"` is true.